src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -314,7 +314,8 @@ public Activity Start()
                     StartTimeUtc = GetUtcNow();
                 }
 
-                if (IdFormat == ActivityIdFormat.W3C || (ParentId == null && DefaultIdFormat == ActivityIdFormat.W3C))
+                var defaultFormat = DefaultIdFormat;
+                if (IdFormat == ActivityIdFormat.W3C || defaultFormat == ActivityIdFormat.ForceW3C || (ParentId == null && defaultFormat == ActivityIdFormat.W3C))
                 {
                     Id = GenerateW3CId();
                     IdFormat = ActivityIdFormat.W3C;
```

Comment:
It seems odd to potentially assign IdFormat = Hierarchial in SetParentId, then reverse the decision after calling Start(). That provides callers an opportunity to observe both values. I suggest that if we are forcing an id format then IdFormat property should either be uninitialized or the forced value at all times.